From 24dbe31c2e9c6788b7a8d271f760a3a3c8c2a0f7 Mon Sep 17 00:00:00 2001
From: Wale Ogunwale <ogunwale@google.com>
Date: Thu, 14 Jun 2018 08:55:42 -0700
Subject: [PATCH 205/306] Make testSplitscreenPortraitAppOrientationRequests
 pass to unblock release

Work around to unblock the release for failing test
ActivityManagerAppConfigurationTests#testSplitscreenPortraitAppOrientationRequests
which shouldn't be failing since home stack shouldn't be visible. We need
to dig deeper to see why it is failing.
NOTE: Not failing on current master...

Bug: 110159357
Test: atest ActivityManagerAppConfigurationTests#testSplitscreenPortraitAppOrientationRequests
Change-Id: I69423eef3661133b2a032a166745157b3e344751
(cherry picked from commit cf2d6475f9fc885e8bc9761ef2d839eaad3f32b4)
---
 .../core/java/com/android/server/wm/DisplayContent.java  | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/wm/DisplayContent.java b/services/core/java/com/android/server/wm/DisplayContent.java
index b0e6208fdff..cd8fdbfbc64 100644
--- a/services/core/java/com/android/server/wm/DisplayContent.java
+++ b/services/core/java/com/android/server/wm/DisplayContent.java
@@ -3538,7 +3538,14 @@ class DisplayContent extends WindowContainer<DisplayContent.DisplayChildWindowCo
                 // docked or freeform stack is visible...except for the home stack/task if the
                 // docked stack is minimized and it actually set something.
                 if (mHomeStack != null && mHomeStack.isVisible()
-                        && mDividerControllerLocked.isMinimizedDock()) {
+                        && mDividerControllerLocked.isMinimizedDock()
+                        // TODO(b/110159357): Work around to unblock the release for failing test
+                        // ActivityManagerAppConfigurationTests#testSplitscreenPortraitAppOrientationRequests
+                        // which shouldn't be failing since home stack shouldn't be visible. We need
+                        // to dig deeper to see why it is failing. NOTE: Not failing on current
+                        // master...
+                        && !(mDividerControllerLocked.isHomeStackResizable()
+                            && mHomeStack.matchParentBounds())) {
                     final int orientation = mHomeStack.getOrientation();
                     if (orientation != SCREEN_ORIENTATION_UNSET) {
                         return orientation;
-- 
2.17.1

